std/typecons.d

```diff
@@ -6434,6 +6434,21 @@ refCountedStore.ensureInitialized). Otherwise, just issues $(D
 assert(refCountedStore.isInitialized)).
  */
     alias refCountedPayload this;
+
+    static if (is(T == struct) && !is(typeof((T t) => t.toString())))
+    {
+        string toString()()
+        {
+            import std.conv : to;
+            return to!string(refCountedPayload);
+        }
+
+        string toString()() const
+        {
+            import std.conv : to;
+            return to!string(refCountedPayload);
+        }
```

Comment:
You only need one. Please either use `inout` or a templated `this`.

Reply:
`inout` doesn't work here (the new unittest will fail) but templated `this` is exactly what I'm looking for.